src/Lorentz/Contracts/RegistryDAO/Types.hs

```diff
@@ -0,0 +1,105 @@
+-- SPDX-FileCopyrightText: 2020 TQ Tezos
+-- SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+-- | RegistryDAO Types
+module Lorentz.Contracts.RegistryDAO.Types
+  ( RegistryDaoContractExtra (..)
+  , RegistryEntry (..)
+  , RegistryUpdate (..)
+  , RegistryDaoProposalMetadata (..)
+  , IsoRegistryDaoProposalMetadata
+  ) where
+
+import Lorentz
+
+import qualified Lorentz.Contracts.BaseDAO.Types as DAO
+
+
+data RegistryDaoContractExtra k v = RegistryDaoContractExtra
+  { ceRegistry :: BigMap k (RegistryEntry v)
+  }
+  deriving stock (Generic)
+
+instance (NiceComparable k, Ord k, IsoValue k, IsoValue v)
+  => (IsoValue (RegistryDaoContractExtra k v))
+
+instance (HasAnnotation k, HasAnnotation v)
+  => HasAnnotation (RegistryDaoContractExtra k v) where
+    annOptions = DAO.baseDaoAnnOptions
+
+instance (Ord k, TypeHasDoc k, TypeHasDoc v)
+  => TypeHasDoc (RegistryDaoContractExtra k v) where
+    typeDocMdDescription =
+      "TODO"
+    typeDocMdReference = poly2TypeDocMdReference
+    typeDocHaskellRep = concreteTypeDocHaskellRep @(RegistryDaoContractExtra ByteString ByteString)
+    typeDocMichelsonRep = concreteTypeDocMichelsonRep @(RegistryDaoContractExtra ByteString ByteString)
+
+instance (Ord k) => Default (RegistryDaoContractExtra k v) where
+  def = RegistryDaoContractExtra mempty
+
+
+data RegistryEntry v = RegistryEntry
+  { reValue :: Maybe v
+   -- optional statistics on given proposal
+  , reLastAgoraPostId :: Natural
+  , reLastUpdated :: Timestamp
+  } deriving stock (Generic)
+  deriving anyclass (IsoValue)
+
+instance (HasAnnotation v)
+  => HasAnnotation (RegistryEntry v) where
+    annOptions = DAO.baseDaoAnnOptions
+
+instance (TypeHasDoc v)
+  => TypeHasDoc (RegistryEntry v) where
+    typeDocMdDescription =
+      "TODO"
+    typeDocMdReference = poly1TypeDocMdReference
+    typeDocHaskellRep = concreteTypeDocHaskellRep @(RegistryEntry ByteString)
+    typeDocMichelsonRep = concreteTypeDocMichelsonRep @(RegistryEntry ByteString)
+
+
+data RegistryDaoProposalMetadata k v = RegistryDaoProposalMetadata
+  { pmAgoraPostId :: Natural
+  , pmDiff :: [RegistryUpdate k v]
+  }
```

Comment:
As mentioned in chat, it should have another constructor to update some parameters.